### PR TITLE
Split large start and stop methods

### LINKED ISSUE #35396: Method is too large in `teststart` method
**Description:**

I am getting the following error when I try to add some ballerina tests to the HTTP module repo : 
```
Compiling source
        ballerina/http_tests:2.2.2
ERROR [http_tests:(2:2,2:2)] method is too large: '.<teststart>'
```

When I removed some of the tests, the test source is getting compiled. This is even happens if I remove some of the code segments.

Also this message is getting printed when the ballerina tests are running (not a recent one, this has been getting printed for quite some time), not sure whether this is related or not :
```
Running Tests with Coverage
        http_tests
java.lang.instrument.IllegalClassFormatException: Error while instrumenting ballerina/http_tests/2/http_tests.
        at org.jacoco.agent.rt.internal_f3994fa.CoverageTransformer.transform(CoverageTransformer.java:94)
        at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:246)
        at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
        at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:563)
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:555)
        at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:458)
        at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:452)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:451)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at org.ballerinalang.test.runtime.BTestRunner.execute(BTestRunner.java:272)
        at org.ballerinalang.test.runtime.BTestRunner.runTest(BTestRunner.java:135)
        at org.ballerinalang.test.runtime.util.TesterinaUtils.executeTests(TesterinaUtils.java:90)
        at org.ballerinalang.test.runtime.Main.startTestSuit(Main.java:119)
        at org.ballerinalang.test.runtime.Main.main(Main.java:100)
Caused by: java.io.IOException: Error while instrumenting ballerina/http_tests/2/http_tests.
        at org.jacoco.agent.rt.internal_f3994fa.core.instr.Instrumenter.instrumentError(Instrumenter.java:160)
        at org.jacoco.agent.rt.internal_f3994fa.core.instr.Instrumenter.instrument(Instrumenter.java:110)
        at org.jacoco.agent.rt.internal_f3994fa.CoverageTransformer.transform(CoverageTransformer.java:92)
        ... 18 more
Caused by: org.jacoco.agent.rt.internal_f3994fa.asm.MethodTooLargeException: Method too large: ballerina/http_tests/2/http_tests.$gen$$0046$0060teststart$0062 (Lio/ballerina/runtime/internal/scheduling/Strand;)Ljava/lang/Object;
        at org.jacoco.agent.rt.internal_f3994fa.asm.MethodWriter.computeMethodInfoSize(MethodWriter.java:2087)
        at org.jacoco.agent.rt.internal_f3994fa.asm.ClassWriter.toByteArray(ClassWriter.java:496)
        at org.jacoco.agent.rt.internal_f3994fa.core.instr.Instrumenter.instrument(Instrumenter.java:91)
        at org.jacoco.agent.rt.internal_f3994fa.core.instr.Instrumenter.instrument(Instrumenter.java:108)
        ... 19 more
Sample products are added.
```

**Steps to reproduce :**
1. Clone the following repo : https://github.com/TharmiganK/module-ballerina-http
2. Checkout to `method-too-large` branch
3. Uncomment the code segments in `module-ballerina-http-response-interceptors/ballerina-tests/tests/interceptor_special_test.bal`
4. Run the build command : `./gradlew clean test -x :http-native:test -x :http-ballerina-tests:startLdapServer -x :http-ballerina-tests:stopLdapServer -x :http-compiler-plugin-tests:test` (Skipping java tests and docker container startup) 
5. You could see the error : `ERROR [http_tests:(2:2,2:2)] method is too large: '.<teststart>'`
6. If you uncomment only one or two tests the tests are getting compiled (there may be some test failures but it will get compiled)

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title
Fixes #35396

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/Constants.java [hunk 1]
```diff
@@ -39,6 +39,7 @@ private Constants() {
     public static final String OPEN_SEALED_ARRAY = "*";
 
     public static final int INIT_METHOD_SPLIT_SIZE = 50;
+    public static final int LISTENER_COUNT_PER_METHOD = 50;
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/Constants.java:42]: Is 50 working. I thought it should be 25?

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/jvm/CodegenErrorsTest.java [hunk 2]
```diff
@@ -73,4 +73,10 @@ public void testLargeMethods() {
         CompileResult result = BCompileUtil.compile("test-src/jvm/largeMethods");
         BRunUtil.invoke(result, "main");
     }
+
+    @Test
+    public void testLargeNumberOfListeners() {
+        CompileResult result = BCompileUtil.compile("test-src/jvm/large-number-of-listeners.bal");
+        Assert.assertEquals(result.getErrorCount(), 0);
+    }
```
Comment (warunalakshitha) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/jvm/CodegenErrorsTest.java:81]: Shall we also run this code as well during test. We can increment global count during start and check whether the count is == listener count during main fucntion

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/LargeMethodSplitter.java [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+    }
+
+    /**
+     * Split package start function into several smaller functions.
+     *
+     * @param packageNode package node
+     * @param env symbol environment
+     * @return initial start function but trimmed in size
+     */
+    private BLangFunction splitStartFunction(BLangPackage packageNode, SymbolEnv env) {
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/LargeMethodSplitter.java:210]: Shall we add a comment with how function is splitted.

Comment (Nadeeshan96) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/LargeMethodSplitter.java:210]: Added comments.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/jvm/CodegenErrorsTest.java (hunk 2 above)
Comment (Nadeeshan96) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/jvm/CodegenErrorsTest.java:84]: Now runs the code during the test.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/Constants.java (hunk 1 above)
Comment (Nadeeshan96) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/Constants.java:42]: I checked again and 50 was working, even with test coverage. When running with no test coverage, method too large came with 220 listeners. When I was showing http tests running with coverage during the meeting, I think there I had split by 100 listeners and that's why that method too large stack trace showed.

Comment (Nadeeshan96) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/Constants.java]: Since 25 is better anyway, changed it to 25.

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**